src/snapshotcloneserver/snapshot/snapshot_core.cpp

```diff
@@ -1159,6 +1144,32 @@ int SnapshotCoreImpl::HandleCancelUnSchduledSnapshotTask(
     return kErrCodeSuccess;
 }
 
+
+int SnapshotCoreImpl::HandleCancelScheduledSnapshotTask(
+    std::shared_ptr<SnapshotTaskInfo> task) {
+    LockGuard lockGuard(task->GetLockRef());
+
+    if (task->IsFinish()) {
+        return kErrCodeCannotCancelFinished;
+    }
+
+    auto ret = StartCancel(task);
```

Comment:
After "UpdateSnapshot" here, maybe another thread is also doing "UpdateSnapshot", so concurrency problems may occur.

Reply:
> After "UpdateSnapshot" here, maybe another thread is also doing "UpdateSnapshot", so concurrency problems may occur.

yeap, i will fix it.